python/common/org/python/types/Object.java

```diff
@@ -32,6 +32,84 @@ void setValue(org.python.Object obj) {
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("==", "__eq__");
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("!=", "__ne__");
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("is", "__eq__");
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("is not", "__ne__");
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("exception match", "__eq__");
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("in", "__contains__");
+        COMPARISON_OPERATORS_TO_FUNCTION_NAME.put("not in", "__not_contains__");
+
+        COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.put("<", "__gt__");
+        COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.put("<=", "__ge__");
+        COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.put(">", "__lt__");
+        COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.put(">=", "__le__");
+        COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.put("==", "__eq__");
+        COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.put("!=", "__ne__");
+    }
+
+    public org.python.Object comparison(org.python.Object other, String operator) {
+        org.python.Object result = org.python.types.NotImplementedType.NOT_IMPLEMENTED;
+        boolean reflectedChecked = false;
+
+        if (this.getClass() != other.getClass() && this.getClass().isInstance(other)) {
+            reflectedChecked = true;
+            result = invokeComparison(other, this, COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.get(operator));
+            if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
+                return result;
+            }
+        }
+
+        result = invokeComparison(this, other, COMPARISON_OPERATORS_TO_FUNCTION_NAME.get(operator));
+        if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
+            return result;
+        }
+
+        if (!reflectedChecked) {
+            result = invokeComparison(other, this, COMPARISON_OPERATORS_TO_REFLECTED_FUNCTION_NAME.get(operator));
+            if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
+                return result;
+            }
+        }
+
+        // 4.3: Objects of different types, except different numeric types, never compare equal.
+        if (operator.equals("==")) {
+            return new org.python.types.Bool(false);
+        } else if (operator.equals("!=")) {
+            // Non-identical instances of a class normally compare as non-equal unless the class defines the __eq__() method.
+            return new org.python.types.Bool(true);
+        }
+        // 4.3: The <, <=, > and >= operators will raise a TypeError exception when comparing a complex number with another built-in numeric type, when the objects are of different types that cannot be compared, or in other cases where there is no defined ordering.
+        throw new org.python.exceptions.TypeError(String.format(
+        "unorderable types: %s() %s %s()", this.typeName(), operator, other.typeName()));
```

Comment:
This should be indented so it looks like a continuation, rather than a new statement.